### ISSUE TITLE
Issue with Webpacker 6 (beta) and print.css

### ISSUE THREAD

**Issue by thomasdarde** [activeadmin/activeadmin#6908]
When configuring activeadmin to use webpacker, and using webpacker 6 an issue arise with activeadmin/print.css

Maybe renaming it to activeadmin_print.css would avoid having a directory structure and simplify this ?

removing this stylesheet fixes the issue.

**Comment by guillaumebesse** [comment]
I have the same problem (webpacker 6 beta 6).

**Comment by deivid-rodriguez** [comment]
Could any of you investigate this issue a bit more? What kind of issue are you having? Does activeadmin raise an error? Which change in webpacker caused it?

**Comment by DarkSwoop** [comment]
Hi!

I found a workaround that works for now.

I put this at the bottom of my `config/initializers/active_admin.rb`:
```
Rails.application.config.after_initialize do
  ActiveAdmin.application.stylesheets.delete('active_admin/print.css')
  ActiveAdmin.application.stylesheets['active_admin/print'] = { media: 'print' }
end
```

What's happening?
When webpacker tries to look up a pack, it sanitizes the name of the referenced pack with `File.basename`. `File.basename` removes the path before the filename and returns the filename without the extension: `active_admin/print.css` => `print`. And since activeadmin references `active_admin/print.css` webpacker can't find it in the manifest.

The fix for now:
Remove `active_admin/print.css` stylesheet from the stylesheets registry and register it again without the file extension.

Why does this work:
Webpacker returns the name of the asset unchanged when no extension is in the filename.
(https://github.com/rails/webpacker/blob/1258698ccc254bc85c0272dba6de03b766913816/lib/webpacker/manifest.rb#L93)

**Comment by thomasdarde** [comment]
Great to see this investigation ! I did try the workaround with no success (I added this just before `ActiveAdmin.setup do |config|`)

**Comment by DarkSwoop** [comment]
Okay. A little bit more context:

```
# directory structure:
app/javascript:
  packs:
    active_admin.js
    active_admin:
      print.scss
  stylesheets:
    active_admin.scss

ruby:
  activeadmin 2.9.0
  rails 6.1.3.1
  webpacker 6.0.0.beta.7

javascript:
  webpack-dev-server 3.11.2
  webpack 5.35.0
```

**Comment by javierjulio** [comment]
This has bothered me for a long time and its one of the items I resolved in #3862. There is no need for a print stylesheet to be loaded as a separate file. It's been best practice for a long, long time now to just include the print media block `@media print { ... }` in the main stylesheet. I'd much rather we make that change then do anything with Webpacker.

```
/* ==========================================================================
   Print styles.
   Inlined to avoid the additional HTTP request:
   https://www.phpied.com/delay-loading-your-print-css/
   ========================================================================== */

@media print {
```
From: https://github.com/h5bp/html5-boilerplate/blob/master/dist/css/style.css#L189-L195

**Comment by deivid-rodriguez** [comment]
@javierjulio I extracted this idea from #3862 into https://github.com/activeadmin/activeadmin/pull/6922.
